examples/csharp/src/client.cs

```diff
@@ -0,0 +1,136 @@
+using RestSharp;
+using System.Collections.Generic;
+
+public class Client
+{
+    protected string _endpoint;
+    protected bool _selfSigned;
+    protected Dictionary<string, string> _globalHeaders;
+
+    public Client()
+    {
+        _endpoint = "https://appwrite.io/v1";
+        _globalHeaders = new Dictionary<string, string> { "content-type"="", "x-sdk-version"= "appwrite:csharp:" };
+        _selfSigned = false;
+    }
+
+    public SetSelfSigned(bool status = true)
+    {
+        _selfSigned = status;
+    }
+
+    public SetEndpoint(string endpoint)
+    {
+        _endpoint = endpoint;
+    }
+
+    public AddHeader(string key, string value)
+    {
+        if (_globalHeaders.ContainsKey(key.ToLower()))
```

Comment:
Maybe avoid calling ToLower() 1+ times? You could store the result in a local variable or overwrite the value of key?